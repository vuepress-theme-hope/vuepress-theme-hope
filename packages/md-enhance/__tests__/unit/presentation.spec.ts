import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { presentation } from "../../src/node/markdown-it/index.js";

const demo = `
## Slide 1

A paragraph with some text and a [link](https://mrhope.site)

---

## Slide 2

- Item 1
- Item 2

---

## Slide 3.1

\`\`\`js
const a = 1;
\`\`\`

--

## Slide 3.2

$$
J(\\theta_0,\\theta_1) = \\sum_{i=0}
$$
`;

describe("presentation", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(presentation);

  it("Should render", () => {
    const renderResult = markdownIt.render(`
@slidestart
${demo}
@slideend
`);

    expect(renderResult).toMatch(
      /<Presentation id="presentation-.*?" code=".*?" theme=".*?"><\/Presentation>/
    );
    expect(renderResult).toMatchSnapshot();
  });

  it("Should not render", () => {
    expect(
      markdownIt.render(`
${demo}
`)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
@slidestar
${demo}
@slideend
`)
    ).toMatchSnapshot();
  });
});
