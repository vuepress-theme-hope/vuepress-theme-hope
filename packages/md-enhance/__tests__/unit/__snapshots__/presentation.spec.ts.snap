// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`presentation > Should not render 1`] = `
"<h2>Slide 1</h2>
<p>A paragraph with some text and a <a href=\\"https://mrhope.site\\">link</a></p>
<hr>
<h2>Slide 2</h2>
<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<hr>
<h2>Slide 3.1</h2>
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
<p>--</p>
<h2>Slide 3.2</h2>
<p>$$
J(\\\\theta_0,\\\\theta_1) = \\\\sum_{i=0}
$$</p>
"
`;

exports[`presentation > Should not render 2`] = `
"<p>@slidestar</p>
<h2>Slide 1</h2>
<p>A paragraph with some text and a <a href=\\"https://mrhope.site\\">link</a></p>
<hr>
<h2>Slide 2</h2>
<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<hr>
<h2>Slide 3.1</h2>
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
<p>--</p>
<h2>Slide 3.2</h2>
<p>$$
J(\\\\theta_0,\\\\theta_1) = \\\\sum_{i=0}
$$</p>
<p>@slideend</p>
"
`;

exports[`presentation > Should render 1`] = `"<Presentation id=\\"presentation-0\\" code=\\"eJxVjc0KwjAQhO/7FAvtoQX7pzelB4969WilDXYx0aYNyYqC+O4mUJDeho9vZiCK8DSonrAC2KMRVtysMBJfiiW6SRMyvRnF2KPA86DGxyWRzMZti0JbORnKnWJKAbIsg//c2gM8MGk/PIeAFs4m96dd190dXKfR+RessdoFFNSF6ctxDMekYUks2nI1hyr1ncY9dftRdfkN0g8aMTop\\" theme=\\"auto\\"></Presentation>"`;
